test-suite/testkit/tests/counter/main.rs

```diff
@@ -1052,3 +1053,175 @@ fn test_explorer_add_transaction_with_invalid_transaction() {
         ApiError::BadRequest(ref body) if body == error_body
     );
 }
+
+#[test]
+fn test_explorer_api_with_before_transactions_error() {
+    let (mut testkit, api) = init_testkit();
+    let (pubkey, key) = crypto::gen_keypair();
+    let tx = Increment::new(13).sign(SERVICE_ID, pubkey, &key);
+
+    // This tx lead to error in before_transaction on the next transaction
+    testkit.create_block_with_transaction(tx.clone());
+
+    let tx = Increment::new(1).sign(SERVICE_ID, pubkey, &key);
+    // So perform one more tx to check the error
+    let block = testkit.create_block_with_transaction(tx.clone());
+
+    let response = api
+        .public(ApiKind::Explorer)
+        .query(&CallStatusQuery {
+            hash: tx.object_hash(),
+            call_kind: None,
+        })
+        .get::<CallStatusResponse>("v1/call_status")
+        .expect("Transaction send to finish with error.");
```

Comment:
I don't quite understand the meaning of "Transaction send to finish with error.".